server/service/hosts.go

```diff
@@ -837,20 +840,63 @@ type hostsReportRequest struct {
 	Opts    fleet.HostListOptions `url:"host_options"`
 	LabelID *uint                 `query:"label_id,optional"`
 	Format  string                `query:"format"`
+	Columns string                `query:"columns,optional"`
 }
 
 type hostsReportResponse struct {
-	Hosts []*fleet.Host `json:"-"` // they get rendered explicitly, in csv
-	Err   error         `json:"error,omitempty"`
+	Columns []string        `json:"-"` // used to control the generated csv
```

Comment:
Nit: Add a comment `see hijackRender` (at first sight, it's a bit confusing a response type contains the provided columns).